### ISSUE TITLE
Hints no longer applicable for 9.0.1

### ISSUE THREAD

**Issue by jrp2014** [ndmitchell/hlint#1248]
In 9.0.1, if I have

```haskell
import Control.Monad.Catch
import Control.Monad.Cont as Cont
import Control.Monad.Fail as Fail

newtype AnyContT m a = AnyContT { unAnyContT :: forall r . ContT r m a }
```
I seem to need to change  $ . combinations (eg `  fmap f p = AnyContT $ fmap f ( unAnyContT p )` instead of `fmap f p = AnyContT $ fmap f . unAnyContT $ p`.  So

```
  AnyContT f >>= k = AnyContT ( f >>= ( \x -> unAnyContT (k x) ) )
```
compiles OK, but if I follow hlint's hint to get rid of the lambda, I get
```
src/Control/Monad/Trans/AnyCont.hs:25:41: error:
    • Couldn't match type: forall r1. ContT r1 m b
                     with: ContT r m b
      Expected: AnyContT m b -> ContT r m b
        Actual: AnyContT m b -> forall r. ContT r m b
    • In the first argument of ‘(.)’, namely ‘unAnyContT’
      In the second argument of ‘(>>=)’, namely ‘(unAnyContT . k)’
      In the first argument of ‘AnyContT’, namely
        ‘(f >>= (unAnyContT . k))’
    • Relevant bindings include
        k :: a -> AnyContT m b
          (bound at src/Control/Monad/Trans/AnyCont.hs:25:18)
        f :: forall r. ContT r m a
          (bound at src/Control/Monad/Trans/AnyCont.hs:25:12)
        (>>=) :: AnyContT m a -> (a -> AnyContT m b) -> AnyContT m b
          (bound at src/Control/Monad/Trans/AnyCont.hs:25:14)
   |
25 |   AnyContT f >>= k = AnyContT ( f >>= ( unAnyContT . k ) )
```

**Comment by ndmitchell** [comment]
GHC's rules for where you have to introduce lambdas around higher-rank types change pretty regularly. As the user manual for HLint says:

> Some transformed programs may require additional type signatures, particularly if the transformations trigger the monomorphism restriction or involve rank-2 types.

I think if you name the variable `mr` then HLint won't suggest removing it. Otherwise, just supress the warning is probably the best thing to do.
